azure/code/rapids_csp_azure.py

```diff
@@ -317,13 +327,25 @@ def fit_xgboost(self, X_train, y_train, model_params):
         Trains a XGBoost model on X_train and y_train with model_params
 
         Parameters and Objects returned are same as trained_model
-        """
+        """             
+        if 'GPU' in self.compute_type:
```

Comment:
'in' <string> issue.

Reply:
These are checking for single-GPU, multi-GPU and CPUs, so it's not equal and checking whether it's within the string